packages/starlight/components/CallToAction.astro

```diff
@@ -1,17 +1,19 @@
 ---
+import type { HTMLAttributes } from 'astro/types';
 import Icon from '../user-components/Icon.astro';
 import type { Icons } from './Icons';
 
 interface Props {
 	variant: 'primary' | 'secondary' | 'minimal';
 	link: string;
 	icon?: undefined | { type: 'icon'; name: keyof typeof Icons } | { type: 'raw'; html: string };
+	attrs?: HTMLAttributes<"a">;
 }
 
-const { link, variant, icon } = Astro.props;
+const { link, variant, icon, attrs } = Astro.props;
```

Comment:
Ah, follow-up sorry. I just tested again and actually the `class` and `class:list` merging only seems to work when both are declared directly (i.e. `<a class:list={[]} class="">`), not when `class` is in a spread object.

So we’ll need to special case that one:
```suggestion
const { link, variant, icon } = Astro.props;
const { class: customClass, ...attrs } = Astro.props.attrs;
```